pkg/rehearse/jobs.go

```diff
@@ -589,28 +589,22 @@ func getAllAncestors(changed []registry.Node) []registry.Node {
 	return ancestors
 }
 
-func SelectJobsForChangedRegistry(regSteps []registry.Node, allPresubmits presubmitsByRepo, allPeriodics []prowconfig.Periodic, configPath string, loggers Loggers) (config.Presubmits, config.Periodics) {
+func SelectJobsForChangedRegistry(regSteps []registry.Node, allPresubmits presubmitsByRepo, allPeriodics []prowconfig.Periodic, ciopConfigs config.DataByFilename, loggers Loggers) (config.Presubmits, config.Periodics) {
 	// TODO(muller): Improve logging in this method
-	// TODO(muller): Get rid of reading the config here, it should be read at least once already (get rid of the error, too)
-	selectedPresubmits := config.Presubmits{}
-	selectedPeriodics := config.Periodics{}
-	var configs []*config.DataWithInfo
-	if err := config.OperateOnCIOperatorConfigDir(configPath, func(cfg *api.ReleaseBuildConfiguration, info *config.Info) error {
-		configs = append(configs, &config.DataWithInfo{Configuration: *cfg, Info: *info})
-		return nil
-	}); err != nil {
-		loggers.Debug.Errorf("Failed to load config by filename in SelectJobsForChangedRegistry: %v", err)
-		return selectedPresubmits, selectedPeriodics
+	// We need a sorted index of ci-operator configs for deterministic behavior
+	var sortedConfigs []*config.DataWithInfo
+	for idx := range ciopConfigs {
+		cfg := ciopConfigs[idx]
+		sortedConfigs = append(sortedConfigs, &cfg)
```

Comment:
IIRC this is going to be putting the pointers to the location in the `ciopConfigs` slice into `sortedConfigs` ... can we get `config.DataByFilename` to have pointers itself inside?

Reply:
I did it intentionally as a slice of pointers - there's no need to copy full structs around and the configs are just read, never written. The only point of the slice is to get a sorted view of the map, so pointers seemed like a reasonably lightweight way to do that.

`config.DataByFilename`  having pointers sounds risky, the struct is afaik used a lot elsewhere and we could get some undesired modifications? But I can try it and see what happens.

Reply:
I tried to change `config.DataByFilename` to work with pointers right from the start, but the change is quite invasive -- there is code in other tools that relies on zero values from that map, working around that does not seem to be worth the effort, IMO.